crates/fuel-gas-price-algorithm/src/v1/tests/update_da_record_data_tests.rs

```diff
@@ -283,3 +284,54 @@ fn update__da_block_updates_projected_total_cost_with_known_and_guesses_on_top()
     let expected = new_known_total_cost + guessed_part;
     assert_eq!(actual, expected as u128);
 }
+
+#[test]
+fn update_da_record_data__when_reward_is_greater_than_cost_will_zero_reward_and_subtract_from_cost(
```

Comment:
Need to check the projected cost as well, specifically when it is ahead of DA still after the update.